Sources/NIOWebSocket/Base64.swift

```diff
@@ -12,75 +12,122 @@
 //
 //===----------------------------------------------------------------------===//
 
-// The base64 unicode table.
-private let base64Table: [Unicode.Scalar] = [
-    "A", "B", "C", "D", "E", "F", "G", "H",
-    "I", "J", "K", "L", "M", "N", "O", "P",
-    "Q", "R", "S", "T", "U", "V", "W", "X",
-    "Y", "Z", "a", "b", "c", "d", "e", "f",
-    "g", "h", "i", "j", "k", "l", "m", "n",
-    "o", "p", "q", "r", "s", "t", "u", "v",
-    "w", "x", "y", "z", "0", "1", "2", "3",
-    "4", "5", "6", "7", "8", "9", "+", "/",
-]
+// This is a simplified vendored version from:
+// https://github.com/fabianfett/swift-base64-kit
 
 extension String {
-    /// Base64 encode an array of UInt8 to a string, without the use of Foundation.
-    ///
-    /// This function performs the world's most naive Base64 encoding: no attempts to use a larger
-    /// lookup table or anything intelligent like that, just shifts and masks. This works fine, for
-    /// now: the purpose of this encoding is to avoid round-tripping through Data, and the perf gain
-    /// from avoiding that is more than enough to outweigh the silliness of this code.
-    init(base64Encoding array: Array<UInt8>) {
-        // In Base64, 3 bytes become 4 output characters, and we pad to the nearest multiple
-        // of four.
-        var outputString = String()
-        outputString.reserveCapacity(((array.count + 2) / 3) * 4)
 
-        var bytes = array.makeIterator()
-        while let firstByte = bytes.next() {
-            let secondByte = bytes.next()
-            let thirdByte = bytes.next()
-            outputString.unicodeScalars.append(String.encode(firstByte: firstByte))
-            outputString.unicodeScalars.append(String.encode(firstByte: firstByte, secondByte: secondByte))
-            outputString.unicodeScalars.append(String.encode(secondByte: secondByte, thirdByte: thirdByte))
-            outputString.unicodeScalars.append(String.encode(thirdByte: thirdByte))
-        }
+  /// Base64 encode a collection of UInt8 to a string, without the use of Foundation.
+  ///
+  /// This function performs the world's most naive Base64 encoding: no attempts to use a larger
+  /// lookup table or anything intelligent like that, just shifts and masks. This works fine, for
+  /// now: the purpose of this encoding is to avoid round-tripping through Data, and the perf gain
+  /// from avoiding that is more than enough to outweigh the silliness of this code.
+  @inlinable
```

Comment:
all the `@inlinable` in this file don't do anything because this isn't `public` and always consumed within the `NIOWebSocket` module so I think we should take them out.